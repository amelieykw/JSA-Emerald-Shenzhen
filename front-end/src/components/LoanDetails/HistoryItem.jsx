import React from 'react'
import { View } from 'react-native'
import { Text, Icon } from 'native-base'
import PropTypes from 'prop-types'

import style from './HistoryItemStyle'

const HistoryItem = ({ historyItem }) => {
  return (
    <View style={style.body}>
      <View style={[style.circle, historyItem.amount > 0 ? style.up : style.down]}>
        <Icon
          type="Feather"
          name={`trending-${historyItem.amount > 0 ? 'up' : 'down'}`}
          style={style.icon}
        />
      </View>
      <View>
        <Text>{historyItem.text}</Text>
        <Text style={style.date}>{historyItem.date}</Text>
      </View>
    </View>
  )
}

HistoryItem.propTypes = {
  historyItem: PropTypes.shape({
    amount: PropTypes.number,
    text: PropTypes.string,
    date: PropTypes.string
  }).isRequired
}

export default HistoryItem
