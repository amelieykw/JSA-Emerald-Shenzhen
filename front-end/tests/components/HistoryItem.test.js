import React from 'react'
import renderer from 'react-test-renderer'
import HistoryItem from '../../src/components/LoanDetails/HistoryItem'

const mockHistoryItemData = { amount: 4, text: 'text', date: '2019-01-01' }

describe('<HistoryItem />', () => {
  it('<HistoryItem > renders correctly', () => {
    const tree = renderer.create(<HistoryItem historyItem={mockHistoryItemData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
