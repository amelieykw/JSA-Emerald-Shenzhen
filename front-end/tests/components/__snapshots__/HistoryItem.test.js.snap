// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryItem /> <HistoryItem > renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginVertical": 5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 23,
          "height": 45,
          "justifyContent": "center",
          "marginRight": 20,
          "width": 45,
        },
        Object {
          "backgroundColor": "#FC7043",
        },
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      text
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "grey",
            "fontSize": 15,
          },
        ]
      }
      uppercase={false}
    >
      2019-01-01
    </Text>
  </View>
</View>
`;
